### DISCUSSION TITLE
Favicon Issue - Not showing (v9.4.4)

### DISCUSSION THREAD

**Discussion by grumpyTofu** [vercel/next.js#13985] [Help] [answered]
# Bug report

## Describe the bug

Favicon is not showing

## To Reproduce

Insert `<link rel="icon" type="image/x-icon" href="/public/favicon.ico" />` in the custom _document.js file within the `<Head></Head>` tags

## Expected behavior

icon should be displayed in tab

## System information

- Browser: chrome
- Version of Next.js: 9.4.4

## Additional Notes

No errors in console

here is the html head that is produced:

```
<head>
  <link rel="icon" type="image/x-icon" href="/public/favicon.ico">
  <meta charset="utf-8">
  ...
</head>
```

**Comment by mattcarlotta** [comment, marked as answer]
Not sure which `head` you're using, but you're not supposed to import `next/head` within `_document`. Soon, it'll throw an error: #13958 

With [static file serving](https://nextjs.org/docs/basic-features/static-file-serving), you won't need to include `/public`, instead it'll just be `/`.

So, you'll either want to import/utilize `Head` from `next/document` within `_document.js`.

For example:
```js
import Document, { Html, Head, Main, NextScript } from "next/document";
import { version } from "../package.json";

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx);
    return { ...initialProps };
  }

  render() {
    return (
      <Html>
        <Head>
          <meta charSet="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          <meta name="theme-color" content="#000000" />
          <meta name="description" content="Official website for... something." />
          <meta name="build version" content={version} /> 
          <link rel="icon" href="/favicon.ico" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
```
or...

You'll want to import/utilize `Head` from `next/head` within `_app.js`.

For example:
```js
import Head from "next/head";
import { version } from "../package.json";

const App = ({ Component, pageProps }) => ( 
  <>
    <Head>
      <meta charSet="utf-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <meta name="theme-color" content="#000000" />
      <meta name="description" content="Official website for... something." />
      <meta name="build version" content={version} /> 
      <link rel="icon" href="/favicon.ico" />
    </Head> 
    <Component { ...pageProps }/> 
  </>
);


export default App;
```

**Reply by KarthikeyanM28** [reply]
This answer is marked as Correct but the below solution only works

**Comment by bryanprimus** [comment]
have the exact problem but it won't work

**Comment by JeremyAndress** [comment]
This works for me 
```ts
import Head from 'next/head';
import favicon from '../public/favicon.ico';

type Props = {
  title?: string;
};

const CustomHead = ({ title = 'Lecket' }: Props) => (
  <Head>
    <link rel="shortcut icon" href={favicon.src} type="image/x-icon" />
    <title>{title}</title>
  </Head>
);

export default CustomHead;

```

**Reply by andre-nk** [reply]
Thanks, this is perfectly working for me!

**Reply by emarty-maze** [reply]
This should be the accepted answer.

**Reply by longfellowone** [reply]
Not sure why the default starter does not come wired correctly, but this worked for me. Thanks!

**Reply by samcn26** [reply]
face same problem, use this clever solution works perfect. haha

**Reply by muleA** [reply]
thanks,Definitely working

**Reply by KarthikeyanM28** [reply]
Perfect!, Searcing a solution from version 9. now I found it. This should be the accepted answer.

**Reply by TheodoreClemens** [reply]
Genius only thing that worked!

**Reply by rupali-codes** [reply]
I dont know how...but it works for me :)

**Comment by klinsc** [comment]
For people who uses static serving and is facing this issues in nextjs12.
You don't need to add any line of code.
Just place your "favicon.ico" in the root of "public" folder.

Then 
`next build`
and for windows people
`cross-env NODE_ENV=production node server.js` 
or for linux people
`set NODE_ENV=production && node server.js`

Hints: You can use chrome dev tools to help debug the favicon, check it out in the application tab ☝🏻

That's it!
